tensorflow/core/framework/tensor.cc

```diff
@@ -755,15 +756,18 @@ Status Tensor::BitcastFrom(const Tensor& other, DataType dtype,
   shape_.set_data_type(dtype);
   if (buf_ != other.buf_) {
     UnrefIfNonNull(buf_);
-    Tensor ts_ = tensor::DeepCopy(other);
     if (port::kLittleEndian) {
       buf_ = other.buf_;
+      RefIfNonNull(buf_);
```

Comment:
did the RefIfNonNull need to be called in the if/else scopes? Since they are always being called - can we leave them outside the conditional as before?

Reply:
Hi @sagunb ,

Yes. I've tried to move `RefIfNonNull(buf_)` out of the conditional block before. However, it didn't work due to the object scope of Tensor instance `ts_`. If `RefIfNonNull(buf_)` isn't in the same scope as `ts_`, tensor data couldn't be retained in `buf_` correctly, so I have to call `RefIfNonNull(buf_)` in both if/else scopes.